src/sql/workbench/contrib/query/browser/keyboardQueryActions.ts

```diff
@@ -210,18 +213,24 @@ export class RunCurrentQueryWithActualPlanKeyboardAction extends Action {
 	public static ID = 'runCurrentQueryWithActualPlanKeyboardAction';
 	public static LABEL = nls.localize('runCurrentQueryWithActualPlanKeyboardAction', "Run Current Query with Actual Plan");
 
+	private queryPlanTelemetry: QueryPlanTelemetryHelper;
+
 	constructor(
 		id: string,
 		label: string,
-		@IEditorService private _editorService: IEditorService
+		@IEditorService private _editorService: IEditorService,
+		@IInstantiationService instantiationService: IInstantiationService
 	) {
 		super(id, label);
 		this.enabled = true;
+		this.queryPlanTelemetry = instantiationService.createInstance(QueryPlanTelemetryHelper);
 	}
 
 	public override run(): Promise<void> {
 		const editor = this._editorService.activeEditorPane;
 		if (editor instanceof QueryEditor) {
+			let planOptions = { displayActualQueryPlan: true } as ExecutionPlanOptions;
+			this.queryPlanTelemetry.addTelemetry(TelemetryKeys.TelemetryAction.ViewExecutionPlan, editor.input.uri, planOptions);
```

Comment:
This isn't really "viewing" the plan though - it's just running the current query and telling it to return the actual query plan as well. 

So these almost seem like something that should instead go in the connection-related telemetry instead, where every time a query is ran we include the properties `displayActualQueryPlan` and `displayEstimatedQueryPlan`.

(and in addition you should be ensuring you have separate telemetry for when we actual display the editor. There should already be events for specific editors being displayed so you may not need to do anything specific there unless you wanted to add in some custom properties)

Reply:
Do I need to define a connection oriented enum for this kind of telemetry or is there one already available for use? 

I'm going to adding additional telemetry in a second PR, but this was just to ensure that we had telemetry around actual and estimated execution plans before the rest are brought in.

Reply:
We have events sent for every connection, I can't recall exactly where those are done. I'd suggest setting a breakpoint in the telemetry service and then making a connection to see where we're sending events and whether you can hook into that somehow.

Reply:
@Charles-Gagnon, I guess you meant to say query related? I think we should have an event named: ExecuteQuery, and we can add a property: executionPlanType to it. 

@lewis-sanchez telemetry should be carefully reviewed, if you change the report data structure, the telemetry report will need to be adjusted.

Reply:
Correct @alanrenmsft - we have an event for `RunQuery` already so we should look into adding the additional properties there

https://github.com/microsoft/azuredatastudio/blob/02341088ebde4e88b5e84e534bf70e3506ded54f/src/sql/workbench/services/query/common/queryManagement.ts#L233

Reply:
And that already has the `ExeuctionPlanOptions` passed in so that should be easy to add there.